twisted-infra/braid
# Fix installing private data in buildbot fab definition.

Problem
======

buildmaster private date for stating/testing is a duplicate of the actual private data and get out of sync.


Changes
=======

Update private data with latest buildslaves which are present in production.

As a drive by I have updated the base service code to create the config folder.

Instead of using installPrivateData and installTestData, I just want for installPrivateData ... when it is False it will install the non-private stuff.

For now we need to manually keep in sync the production and stating private data... but I hope that now that we have the vagrant vm available it is much easier to detect out of sync databases.

In the future we might think of a way to reduce the duplication.... but for now I have no idea how to do it.


How to test
=========

On the vagrant, make sure /srv/bb-master folder is clean.

Run `fab -i ~/.vagrant.d/insecure_private_key config.vagrant buildbot.install`  and then check http://172.16.255.140:8080/buildslaves .. it should start and all slaves should have a builder
